llvmDebugInfoC/src/scripts/konan_lldb.py

```diff
@@ -37,74 +44,68 @@ def lldb_val_to_ptr(lldb_val):
 
 def evaluate(expr):
     result = lldb.debugger.GetSelectedTarget().EvaluateExpression(expr, lldb.SBExpressionOptions())
-    log("{} => {}".format(expr, result))
+    evallog = lambda : "{} => {}".format(expr, result)
+    log(evallog)
+    exelog(evallog)
     return result
 
 def is_instance_of(addr, typeinfo):
     return evaluate("(bool)IsInstance({}, {})".format(addr, typeinfo)).GetValue() == "true"
 
+def is_string_or_array(value):
+    return evaluate("(bool)IsInstance({0}, theStringTypeInfo) ? 1 : ((int)Konan_DebugIsArray({0}) ? 2 : 0)".format(lldb_val_to_ptr(value))).unsigned
 
-def is_string(value):
-    return is_instance_of(lldb_val_to_ptr(value), "theStringTypeInfo")
-
-
-def is_array(value):
-    return evaluate("(int)Konan_DebugIsArray({})".format(lldb_val_to_ptr(value))).unsigned == 1
-
-
-def check_type_info(value):
+def type_info(value):
     """This method checks self-referencing of pointer of first member of TypeInfo including case when object has an
     meta-object pointed by TypeInfo. Two lower bits are reserved for memory management needs see runtime/src/main/cpp/Memory.h."""
     if str(value.type) != "struct ObjHeader *":
         return False
-    expr = "*(void **)((uintptr_t)(*(void**){0:#x}) & ~0x3) == **(void***)((uintptr_t)(*(void**){0:#x}) & ~0x3)".format(value.unsigned)
+    expr = "*(void **)((uintptr_t)(*(void**){0:#x}) & ~0x3) == **(void***)((uintptr_t)(*(void**){0:#x}) & ~0x3) ? *(void **)((uintptr_t)(*(void**){0:#x}) & ~0x3) : (void *)0".format(value.unsigned)
```

Comment:
Not sure if it shall be obfuscated C contest.

Reply:
I've just combined two calls  old type_info and check_type_info to avoid extra compilations.